### DISCUSSION TITLE
how to use this part of js code ?

### DISCUSSION THREAD

**Discussion by SunlearnDev** [mdbootstrap/TW-Elements#1773] [Chat about dev stuff]
https://tailwind-elements.com/docs/standard/components/carousel/

I'm using tailwind-elements for the first time, I'm confused as to where this js code is used.
// Initialization for ES Users
import {
  Carousel,
  initTE,
} from "tw-elements";

initTE({ Carousel });

**Comment by ascensus-mdb** [comment]
This is so-called initialization. Have a look at our tutorials, where everything is explained

https://tailwind-elements.com/learn/te-foundations/basics/vite/

**Comment by dristy19** [comment]
same problem... did u get the solution??

**Reply by ascensus-mdb** [reply]
Have you seen the answer above?

If you are using a different installation method than CDN then you need to initialize the components for them to work. This will keep your project optimized and only use the code you need.

See the tutorial I wrote above, it's all explained there.

**Reply by TBemrose** [reply]
@ascensus-mdb
That page doesn't explain the "if you are using a different installation method than CDN" bit you just did, which is critical to answering the question. (Nor does it explain whatever an "ES module" is.)

**Reply by juujisai** [reply]
Hi @TBemrose! The `initTE` method is needed when we want to add a auto initialization for used components. TW Elements provides two formats - `UMD` - where `initTE` is beeing called automaticaly at the beginning and `ES` where we have to call the method ourselves. ES format will be used in most apps where we define - `"type": "module"` inside the `package.json` file.

We have some description for the UMD/ES formats on our Quick Start page:
https://tw-elements.com/docs/standard/getting-started/quick-start/#formats

And Optimization page:
https://tw-elements.com/docs/standard/getting-started/optimization/#modules
